api/client/client.go

```diff
@@ -24,10 +25,51 @@ type Client struct {
 	conn   *grpc.ClientConn
 }
 
+// ActionType describes the type of event action when subscribing to data updates
+type ActionType int
+
+const (
+	// ActionCreate represents an event for creating a new entity
+	ActionCreate ActionType = iota + 1
+	// ActionSave represents an event for saving changes to an existing entity
+	ActionSave
+	// ActionDelete represents an event for deleting existing entity
+	ActionDelete
+)
+
+// Action represents a data event delivered to a listener
+type Action struct {
+	Model    string
+	Type     ActionType
+	EntityID string
+	Entity   []byte
+}
+
+// ListenActionType describes the type of event action when receiving data updates
+type ListenActionType int
+
+const (
+	// ListenAll specifies that Create, Save, and Delete events should be listened for
+	ListenAll ListenActionType = iota
+	// ListenCreate specifies that Create events should be listened for
+	ListenCreate
+	// ListenSave specifies that Save events should be listened for
+	ListenSave
+	// ListenDelete specifies that Delete events should be listened for
+	ListenDelete
+)
+
+// ListenOption represents a filter to apply when listening for data updates
+type ListenOption struct {
+	Type     ListenActionType
+	Model    string
+	EntityID string
+}
+
```

Comment:
Modeled this stuff in the client to avoid having to deal with the friction of using core `EntityID` type that is used on the server.

Reply:
Nice, makes sense. Always feels a little odd to use server models in a client, even if you _could_.